### PR TITLE
Add config flag to specify runtime config files

### PR DESCRIPTION
**Pull request by Matt-Esch**
I have removed the hard-coded configurations for each binary. Now the defaults are bundled with the runtime, and extra configs are passed with a --config flag to the binary.

### CODE REVIEW

File: config/production.json.go [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+	"metrics.runtime.enableMemMetrics": true,
+	"metrics.runtime.enableGCMetrics": true,
+	"metrics.runtime.collectInterval": 1000,
+
+	"tchannel.serviceName": "my-gateway",
+	"tchannel.processName": "my-gateway"
+}
+`)
+
+func productionJsonBytes() ([]byte, error) {
```
Comment (ChuntaoLu) [inline config/production.json.go]: Is this function necessary? Seems like it is only used in the function below and doesn't do much.

File: config/production.json.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+	return fi.modTime
+}
+func (fi bindataFileInfo) IsDir() bool {
+	return false
+}
+func (fi bindataFileInfo) Sys() interface{} {
+	return nil
+}
+
+var _productionJson = []byte(`{
```
Comment (ChuntaoLu) [inline config/production.json.go]: How is this going to be updated? And how do we make sure it's valid json?

Comment (Matt-Esch) [inline config/production.json.go]: It's generated the same way that the template bundle is generated, it's added to the generate makefile. Staleness is caught in tests

File: config/production.json.go (hunk 1 above)
Comment (Matt-Esch) [inline config/production.json.go]: It's a generated function. It returns the bytes for productionJson, and it's a private function

**Pull request merged**